python/cudf/cudf/dataframe/numerical.py

```diff
@@ -65,8 +65,23 @@ def deserialize(cls, header, frames):
         return col
 
     def binary_operator(self, binop, rhs, reflect=False):
+        int_dtypes = [
+            np.dtype("int8"),
+            np.dtype("int16"),
+            np.dtype("int32"),
+            np.dtype("int64"),
+        ]
+        tmp = rhs
+        if reflect:
+            tmp = self
         if isinstance(rhs, NumericalColumn) or np.isscalar(rhs):
             out_dtype = np.result_type(self.dtype, rhs.dtype)
+            if binop in ["mod", "floordiv"]:
+                if (
+                    (isinstance(tmp, NumericalColumn) and (0 in tmp))
```

Comment:
Doing `0 in tmp` will be insanely expensive if rhs is a big NumericalColumn until we resolve #2243.